WordPress/gutenberg
# Editor Canvas: Fix animation when device type changes

## What?
This PR fixes an awkward animation in the editor canvas when changing device type.

### Before

When switching from Mobile or Tablet to Desktop, the canvas area moves to the top left and then animates.

https://github.com/WordPress/gutenberg/assets/54422211/2edf4fb3-ab16-44d4-a0cf-8c31a6ac5c6c

### After

The left/right center position should be maintained no matter what device type you switch to.

https://github.com/WordPress/gutenberg/assets/54422211/49a0fde4-26cb-45c7-9c38-12b833a67db5

## Why?

Tablets and mobile devices have `margin: auto` on the left and right via the `useResizeCanvas` hook. However, the desktop does not have any margin, so the transition CSS does not work correctly.

## How?

I first thought of adding default left and right auto margins to `useResizeCanvas` itself. However, the documentation [mentions](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/use-resize-canvas/README.md#useresizecanvas) that "it has no effect on the desktop".

Therefore, I added auto margin to the higher-level component, the `EditorCanvas` component.

**Update:** I included margin in the `useResizeCanvas` hook itself. See [here](https://github.com/WordPress/gutenberg/pull/56970#discussion_r1423735744) for the context.

## Testing Instructions

Switch device types to see the animation.

> [!NOTE]
> If the desktop preview of the Post Editor is not iframed, the animation will not work in the first place. This problem is summarized in #56968.
